src/annotation.rs

```diff
@@ -6,125 +6,95 @@
 ///
 /// The borrowed `Max<K>` will be used to define the traversal path over the tree.
-pub trait MapAnnotation<K, V, S>
+pub trait MapAnnotation<K, V>
 where
-    K: Canon<S> + Ord,
-    V: Canon<S>,
-    S: Store,
-    Self: Canon<S> + Annotation<KelvinMap<K, V, Self, S>, S>,
-    Self: Borrow<Max<K>> + Borrow<Cardinality>,
+    K: Canon + Ord,
+    V: Canon,
+    Self: Canon + Annotation<Leaf<K, V>> + Combine<KelvinMap<K, V, Self>, Self>,
+    Self: Borrow<MaxKey<K>> + Borrow<Cardinality>,
 {
 }
 
-#[derive(Debug, Clone, Canon)]
+#[derive(Debug, Clone, Default, Canon)]
 /// Minimum working annotation for the KelvinMap.
 ///
 /// Internally contains a [`Max`] implementation for `K`.
 ///
 /// The `Default` implementation of `K` will be considered as the negative infinity for the `Max`
 /// annotation.
-pub struct MapAnnotationDefault<K, S>
+pub struct MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    S: Store,
+    K: Canon + Ord + Default,
 {
     cardinality: Cardinality,
-    max: Max<K>,
-    store: PhantomData<S>,
+    max: MaxKey<K>,
 }
 
-impl<K, V, S> MapAnnotation<K, V, S> for MapAnnotationDefault<K, S>
+impl<K> Borrow<MaxKey<K>> for MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    V: Canon<S>,
-    S: Store,
+    K: Canon + Ord + Default,
 {
-}
-
-impl<K, S> Borrow<Max<K>> for MapAnnotationDefault<K, S>
-where
-    K: Canon<S> + Ord + Default,
-    S: Store,
-{
-    fn borrow(&self) -> &Max<K> {
+    fn borrow(&self) -> &MaxKey<K> {
         &self.max
     }
 }
 
-impl<K, S> Borrow<Cardinality> for MapAnnotationDefault<K, S>
+impl<K> Borrow<Cardinality> for MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    S: Store,
+    K: Canon + Ord + Default,
 {
     fn borrow(&self) -> &Cardinality {
         &self.cardinality
     }
 }
 
-impl<K, V, S> Annotation<KelvinMap<K, V, MapAnnotationDefault<K, S>, S>, S>
-    for MapAnnotationDefault<K, S>
+impl<K, V> Annotation<Leaf<K, V>> for MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    V: Canon<S>,
-    S: Store,
+    K: Canon + Ord + Default,
 {
-    fn identity() -> Self {
-        let cardinality = <Cardinality as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::identity();
-        let max = Max::Maximum(K::default());
-
-        Self {
-            cardinality,
-            max,
-            store: PhantomData,
-        }
-    }
-
     fn from_leaf(leaf: &Leaf<K, V>) -> Self {
-        let cardinality = <Cardinality as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_leaf(leaf);
-        let max = <Max<K> as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_leaf(leaf);
+        let cardinality =
+            <Cardinality as Annotation<Leaf<K, V>>>::from_leaf(leaf);
+        let max = <MaxKey<K> as Annotation<Leaf<K, V>>>::from_leaf(leaf);
```

Comment:
You should be able to reduce boilerprate here:

```suggestion
        let cardinality =
            Cardinality::from_leaf(leaf);
        let max = MaxKey::from_leaf(leaf);
```

Reply:
Fixed, was using the old required syntax